RELEASING.md

```diff
@@ -0,0 +1,72 @@
+# Releasing Guidelines for Cucumber Godog
+
+This document provides guidelines for releasing new versions of Cucumber Godog. Follow these steps to ensure a smooth and consistent release process.
+
+## Versioning
+
+Cucumber Godog follows [Semantic Versioning]. Version numbers are in the format `MAJOR.MINOR.PATCH`.
+
+- **MAJOR**: Incompatible API changes.
+- **MINOR**: Backward-compatible new features.
+- **PATCH**: Backward-compatible bug fixes.
```

Comment:
This is a common approach for `v1+`, but with current `v0` we're still in officially unstable API mode (`v0` is to avoid import path changes in Go module before we reach stable API). We can not use MAJOR to convey information about backwards compatibility, so a more helpful indication at the moment is to use MINOR for breaking changes and PATCH for all backwards compatible changes.

Reply:
Ok, got it, thanks. How about the below formulation then:

### Current (for v0.MINOR.PATCH)

- **MINOR**: Incompatible API changes.
- **PATCH**: Backward-compatible new features and bug fixes.

### After v1.X.X release

- **MAJOR**: Incompatible API changes.
- **MINOR**: Backward-compatible new features.
- **PATCH**: Backward-compatible bug fixes.